pandas/tests/frame/test_indexing.py

```diff
@@ -185,6 +186,51 @@ def test_setitem_list_of_tuples(self):
         expected = Series(tuples, index=self.frame.index, name='tuples')
         assert_series_equal(result, expected)
 
+    def test_setitem_list_all_missing_columns_scalar(self):
+        df = self.frame.copy()
+        df[['E', 'F']] = 1
+
+        result = df.columns
+        expected = self.frame.columns.union(['E', 'F'])
+        assert_index_equal(result, expected)
+
+        result = df[self.frame.columns]
+        expected = self.frame
+        assert_frame_equal(result, expected)
+
+        assert (df[['E', 'F']] == 1).all().all()
+
+    def test_setitem_list_some_missing_columns_list(self):
+        df = self.frame.copy()
+        df[['A', 'E']] = [1, 2]
+
+        result = df.columns
+        expected = self.frame.columns.union(['E'])
+        assert_index_equal(result, expected)
+
+        result = df[self.frame.columns.drop('A')]
+        expected = self.frame.drop('A', axis=1)
+        assert_frame_equal(result, expected)
+
+        assert (df['A'] == 1).all()
+        assert (df['E'] == 2).all()
+
+    def test_setitem_list_some_missing_columns_dataframe(self):
+        df = self.frame.copy()
+        df[['A', 'E']] = self.frame[['B', 'C']]
+
+        result = df.columns
+        expected = self.frame.columns.union(['E'])
+        assert_index_equal(result, expected)
+
+        result = df[self.frame.columns.drop('A')]
+        expected = self.frame.drop('A', axis=1)
+        assert_frame_equal(result, expected)
+
+        result = df[['A', 'E']]
+        expected = self.frame[['B', 'C']]
+        assert_numpy_array_equal(result.to_numpy(), expected.to_numpy())
```

Comment:
i'm having a hard time just eyeballing these tests. Ideally, should have just one assert per test. construct the expected and use `assert_frame_equal` which checks indexes, values, labels, names etc. with one assert. 

Do let me know if i've misunderstood the intention here.

Reply:
The problem is how to obtain the result `DataFrame`. I suppose one way to do that is to assume the correctness of assigning a single column and obtain the result by assigning a single column multiple times.

Reply:
yes. that sounds good.

Reply:
Updated.